internal/internal_workflow_client.go

```diff
@@ -293,12 +293,24 @@ func (wc *WorkflowClient) ExecuteWorkflow(ctx context.Context, options StartWork
 // reaches the end state, such as workflow finished successfully or timeout.
 // The current timeout resolution implementation is in seconds and uses math.Ceil(d.Seconds()) as the duration. But is
 // subjected to change in the future.
-func (wc *WorkflowClient) GetWorkflow(_ context.Context, workflowID string, runID string) WorkflowRun {
+func (wc *WorkflowClient) GetWorkflow(ctx context.Context, workflowID string, runID string) WorkflowRun {
 
 	iterFn := func(fnCtx context.Context, fnRunID string) HistoryEventIterator {
 		return wc.GetWorkflowHistory(fnCtx, workflowID, fnRunID, true, enumspb.HISTORY_EVENT_FILTER_TYPE_CLOSE_EVENT)
 	}
 
+	// The ID may not actually have been set - if not, we have to ask the server for info about the workflow execution
+	// and extract run id from there. This is definitely less efficient than it could be if there was a more specific
+	// rpc method for this, or if there were more granular history filters - in which case it could be extracted from
+	// the `iterFn` inside of `workflowRunImpl`
+	if runID == "" {
```

Comment:
I would move this code into workflowRunImpl.GetRunID to save the roundtrip for cases when runID is not being accessed.

Reply:
Makes sense, initially I had tried doing this with the history iterator, but found the events wouldn't contain what I needed. I'll plumb in a way to access the client from inside `GetRunID` -- or perhaps make runID be lazily loaded.

Reply:
Made this lazy-load.